compiler/src/iree/compiler/Codegen/Common/ConcretizePadResultShape.cpp

```diff
@@ -132,15 +132,14 @@ class ConcretizePadResultShapePass final
     : public impl::ConcretizePadResultShapePassBase<
           ConcretizePadResultShapePass> {
 public:
-  LogicalResult initialize(MLIRContext *context) override {
-    config.listener = &listener;
-    return success();
-  }
-
   void runOnOperation() override {
     MLIRContext *context = &getContext();
     auto funcOp = getOperation();
 
+    GreedyRewriteConfig config;
+    ConfigTrackingListener listener;
```

Comment:
I think these two lines should be swapped so that the lifetime of listener ends after the lifetime of `config`

Reply:
Ah, good catch. Done.